urnai/base/savable.py

```diff
@@ -0,0 +1,225 @@
+import os
+import pickle
+import tempfile
+from multiprocessing import Process
+
+#import time
+#from urnai.utils.reporter import Reporter as rp
+
+
+class SavableAttr:
+    def __init__(self, value):
+        self.value = value
+
+
+class Savable:
+    """
+    This interface represents the concept of a class that can be saved to disk.
+    The heir class should define a constant or attribute as a default filename
+	to save on disk.
+    """
+
+    def __init__(self, threaded_saving=False):
+        self.threaded_saving = threaded_saving
+        self.pickle_black_list = []
```

Comment:
`pickle` restricts the type of file that will be saved. I would use a more general term, as we intend to use several types of files.